app/mailers/generic_mailer.rb

```diff
@@ -64,6 +64,8 @@ def mc_only?
   end
 
   def process_user_ids(user_ids)
+    user_ids = User.where('id in (?) and receive_email = TRUE', user_ids)
+                   .pluck(:id)
     if mc_only?
       User.where('id in (?) and mc = TRUE', user_ids).pluck(:id)
```

Comment:
You should change this also. Maybe
```suggestion
      User.where(id: user_ids, mc: true, receive_email: true).pluck(:id)
```